harness/determined/constants.py

```diff
@@ -24,16 +26,39 @@
     "optimizations": DEFAULT_OPTIMIZATIONS,
 }
 
+# Allow users to override the ports using the environment variables, in case
+# there are other processes running on the compute node which are already using
+# our default port. Users will likely set these in their "startup-hook.sh".
+# Below is an example how to set these in the experiment's YAML file for testing.
+#
+# environment:
+#   environment_variables:
+#      - DTRAIN_SSH_PORT=13000
+#      - HOROVOD_GLOO_RENDEZVOUS_PORT=13010
+#      - INTER_TRAIN_PROCESS_COMM_PORT_1=13020
+DTRAIN_SSH_PORT_ENV_VAR = "DTRAIN_SSH_PORT"
```

Comment:
can we avoid this `LONG_NAME_VAR = "LONG_NAME"` and just use the `"LONG_NAME"` string directly?

There's a tiny type-checking benefit to the `LONG_NAME_VAR` symbol if we ever change the name, but that doesn't really matter because:
1. the only sane way for us to change an environment variable involves grepping for all instances of the old name anyway
2. changing the name would break all users who depend on it, so it really can't change anyway.